packages/utils/src/abi_encoder/evm_data_types/method.ts

```diff
@@ -34,6 +34,19 @@ export class MethodDataType extends AbstractSetDataType {
         return value;
     }
 
+    public strictDecode<T>(calldata: string, rules?: DecodingRules): T {
+        const value = super.decode(calldata, rules, this._methodSelector);
```

Comment:
I can't find any tests where we check that `decode()` with a mismatched method selector throws. Do we have one?

Reply:
Great catch -- test added!